cloud/grouping.py

```diff
@@ -90,14 +90,20 @@ def _group_nested_tenant(self, instances_list=None):
     def _group_nested_network(self, instances_list=None):
         groups = {}
 
-        network_list = self.network.get_networks_list()
         search_list = (instances_list if instances_list else
                        self.compute.get_instances_list(
                            search_opts={"all_tenants": True}))
-        for network in network_list:
-            LOG.info('Processing network %s', network['name'])
-            groups[str(network['name'])] = [vm for vm in search_list
-                                            if network['name'] in vm.networks]
+        for instance in search_list:
+            LOG.info('Processing instance %s', instance.name)
+            for (network_name, network_ips) in instance.networks.items():
+                network = self.network.get_network({'ip': network_ips[0]},
```

Comment:
Yes. I follow this rule. The result map is 'network_id' => 'instance'.
But I iterates by instance. I think it's doesn't matter.